cirq/sim/wave_function_simulator.py

```diff
@@ -24,7 +24,8 @@
 from cirq.sim import simulator, wave_function
 
 
-class SimulatesIntermediateWaveFunction(simulator.SimulatesIntermediateState,
+class SimulatesIntermediateWaveFunction(simulator.SimulatesAmplitudes,
+                                        simulator.SimulatesIntermediateState,
```

Comment:
Definitely this is valid, but it kind of defeats the purpose of the interface as an indication that it calculates the amplitude efficiently?  I worry that this encourages using this type of simulator for just amplitudes.

Reply:
I think this is convenient and it allows people to start using the interface immediately with the sparse simulator.